src/bindings/python/flux/util.py

```diff
@@ -663,14 +663,18 @@ def __init__(self, name, initial_dict=None):
         if initial_dict:
             self.dict = dict(initial_dict)
 
-        #  Build config search path in order of reverse precedence
-        #  based on XDG specification:
-        self.searchpath = (
-            ["/etc/xdg/flux"]
-            + get_searchpath("XDG_CONFIG_DIRS", "/flux")
-            + [f"{Path.home()}/.config/flux"]
-            + get_searchpath("XDG_CONFIG_HOME", "/flux")
-        )
+        #  Build config search path in precedence order based on XDG
+        #  specification, starting with XDG_CONFIG_HOME (or ~/.config)
+        confdirs = [os.getenv("XDG_CONFIG_HOME") or f"{Path.home()}/.config"]
+
+        #  Append XDG_CONFIG_DIRS as colon separated path (or /etc/xdg)
+        confdirs += (os.getenv("XDG_CONFIG_DIRS") or "/etc/xdg").split(":")
+
+        #  Append "/flux" to all members of confdirs to build searchpath:
+        self.searchpath = [Path(directory, "flux") for directory in confdirs]
+
+        #  Reorder searchpath into reverse precedence order
+        self.searchpath.reverse()
```

Comment:
it took me awhile to figure out why you did this, b/c to the casual observer it makes no sense why you'd put the dirs in a list then just reverse it (vs just putting it in the list in reverse order).  Perhaps just add note about how its b/c XDG_CONFIG_DIRS has higher precedence with paths listed first.

Reply:
Good point, we could put `[::-1]` at the end of  the line that splits XDG_CONFIG_DIRS to insert in reverse order,  but I found that even more inscrutable. I'll add a comment that reminds readers that colon-separated paths are typically in order of precedence and we want to reverse that here.